src/ray/gcs/gcs_server/test/gcs_placement_group_manager_test.cc

```diff
@@ -306,8 +311,15 @@ TEST_F(GcsPlacementGroupManagerTest, TestRescheduleWhenNodeDead) {
   ASSERT_EQ(finished_placement_group_count, 0);
   ASSERT_EQ(mock_placement_group_scheduler_->placement_groups_.size(), 1);
   auto placement_group = mock_placement_group_scheduler_->placement_groups_.back();
+
+  // Manully assign the node field of the first request bundle, simulation scheduling is
+  // done.
   placement_group->GetMutableBundle(0)->set_node_id(NodeID::FromRandom().Binary());
   placement_group->GetMutableBundle(1)->set_node_id(NodeID::FromRandom().Binary());
+
+  // Mark scheduler done, otherwise, it will find the first request placement group
+  // is rescheduling when reschedule on a dead node.
+  gcs_placement_group_manager_->MarkSchedulingDone();
```

Comment:
Is there any problem of using the OnPlacementGroupCreation* instead of making this method public?

Reply:
This callback should be called when simulation scheduling is done technically.

Reply:
here is the case we want to simulate:

1. there is two request pg registered.
2. first pg start schedule.
3. a node dead and reschedule bundle of first pg.
4. schedule second pg.

so, If we invoke `OnPlacementGroupCreationSuccess()` after step 2, It will immediately invoke `SchedulePendingPlacementGroups()` method to start schedule second pg,  there is no chance to simulate the node dead case.

but, I also think make the `MarkSchedulingDone()` public is not a good way, this ut is just for test reschedule on node dead case, so I think the second pg request is not necessary, I want delete the second pg request and only make sure the first pg will reschedule success in node dead case. or, we can `register` second pg after first pg reschedule success.  @ffbin

Reply:
hi, @rkooo567 , Can we temporarily accept this  solution ?

Reply:
Actually no.. let me think about a better way to test this. Usually “temporaray” solution is never fixed after they are pushed, so I prefer to push the proper solution if possible. It I cannot think of good solution, I will just approve! :)

Reply:
ok~